domain/src/main/java/no/difi/meldingsutveksling/core/EDUCoreFactory.java

```diff
@@ -127,6 +144,127 @@ public EDUCore create(Message message, String senderOrgNr) {
         return eduCore;
     }
 
+    public EDUCore create(ConversationResource cr, Arkivmelding am, byte[] asic) {
+        EDUCore eduCore = createCommon(cr.getSenderId(), cr.getReceiverId());
+        eduCore.setId(cr.getConversationId());
+        eduCore.setMessageType(EDUCore.MessageType.EDU);
+        eduCore.setMessageReference(cr.getConversationId());
+
+        if (am.getMappe().isEmpty() || !(am.getMappe().get(0) instanceof Saksmappe)) {
+            throw new MeldingsUtvekslingRuntimeException(String.format("Mappe in Arkivmelding %s not instance of Saksmappe", cr.getConversationId()));
+        }
+        ObjectFactory of = new ObjectFactory();
+        NoarksakType noarksakType = of.createNoarksakType();
+        Saksmappe sm = (Saksmappe) am.getMappe().get(0);
+        ofNullable(sm.getSaksaar()).map(s -> s.toString()).ifPresent(noarksakType::setSaSaar);
+        ofNullable(sm.getSakssekvensnummer()).map(BigInteger::toString).ifPresent(noarksakType::setSaSeknr);
+        noarksakType.setSaAnsvinit(sm.getSaksansvarlig());
+        noarksakType.setSaAdmkort(sm.getAdministrativEnhet());
+        noarksakType.setSaOfftittel(sm.getOffentligTittel());
+
+        if (!(sm.getBasisregistrering().get(0) instanceof Journalpost)) {
+            throw new MeldingsUtvekslingRuntimeException(String.format("Basisregistrering in Arkivmelding %s not instance of Journalpost", cr.getConversationId()));
+        }
+        JournpostType journpostType = of.createJournpostType();
+        Journalpost jp = (Journalpost)  sm.getBasisregistrering().get(0);
+        ofNullable(jp.getJournalaar()).map(BigInteger::toString).ifPresent(journpostType::setJpJaar);
+        ofNullable(jp.getJournalsekvensnummer()).map(BigInteger::toString).ifPresent(journpostType::setJpSeknr);
+        ofNullable(jp.getJournalpostnummer()).map(BigInteger::toString).ifPresent(journpostType::setJpJpostnr);
+        ofNullable(jp.getJournaldato()).map(XMLGregorianCalendar::toString).ifPresent(journpostType::setJpJdato);
+        ofNullable(jp.getJournalposttype()).map(Journalposttype::value).ifPresent(journpostType::setJpNdoktype);
+        ofNullable(jp.getDokumentetsDato()).map(XMLGregorianCalendar::toString).ifPresent(journpostType::setJpDokdato);
+        ofNullable(sm.getSkjerming()).map(Skjerming::getSkjermingshjemmel).ifPresent(journpostType::setJpUoff);
+
+
+        jp.getKorrespondansepart().forEach(k -> {
+            AvsmotType avsmotType = of.createAvsmotType();
+            avsmotType.setAmNavn(k.getKorrespondansepartNavn());
+            if ("avsender".equals(k.getKorrespondanseparttype().value().toLowerCase())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1157)